instance/models/utils.py

```diff
@@ -482,6 +484,49 @@ def purge(self):
         """
         return self._client.kv.delete(self.prefix, recurse=True)
 
+    def leave(self, force=True):
+        """
+        This method will request the agent to leave the cluster, if the
+        operation wasn't completed successfully then it'll force nodes to
+        leave the cluster if force option is set to True.
+
+        :param force: Will fall back to the force mode if the operation
+                      wasn't completed successfuly.
+        :return: True if the operation completed successfully, False otherwise.
+        """
+        request_path = '/v1/agent/leave'
+        is_left = self._api_request(request_path, method='put')
+
+        if not is_left and force:
+            return self.force_leave()
+
+    def force_leave(self):
+        """
+        This method will force the agent nodes to leave the cluster.
+        """
+        _, nodes = self._client.catalog.nodes()
+        for node in nodes:
+            self._client.agent.force_leave(node['Node'])
```

Comment:
Why do we make every single node leave the cluster in this function? That would be catastrophic. We only want a single node to leave, namely the one we specify. And since the version we're using doesn't have the `get` vs `put` fix yet this should actually just be:

```python
request_path = '/v1/agent/force-leave'
return self._api_request(request_path, method='put')
```

otherwise as it currently is it wouldn't work.

Reply:
Sending a force-leave request using this path only `/v1/agent/force-leave` won't put any node in a left state; we have to send a node name in this API for Consul to start the leave procedure. So what I did is using `settings.OCIM_ID`, that's used for Consul to identify a server, to remove this node. However, this might require some changes in our playbooks in running the consul agent to change the node name from `ansible_nodename` to `ocim_id` [[here](https://github.com/open-craft/ansible-playbooks/blob/master/playbooks/roles/consul/defaults/main.yml#L7)]. The main point I saw here is that our playbooks doesn't set `OCIM_ID` env variable as expected so far.